src/libmongoc/src/mongoc/mongoc-server-monitor.c

```diff
@@ -210,6 +210,110 @@ _server_monitor_append_cluster_time (mongoc_server_monitor_t *server_monitor,
    }
    mc_tpld_drop_ref (&td);
 }
+static bool
+_server_monitor_send_and_recv_hello_opmsg (
```

Comment:
`_server_monitor_send_and_recv_hello_opmsg` and `_server_monitor_send_and_recv_opquery` are completely identical in all respects except for the initialization of RPC data. Consider refactoring the two functions such that initialization of RPC data for OP_MSG vs. OP_QUERY is handled separately from the actual "send and recv" implementation. (Related: see implementation of `_mongoc_async_cmd_init_send`.)

Reply:
I entirely agree. I also really, really don't want to touch anything that risks breaking this right now. But it definitely isn't needed and possibly may even cause extra confusion-- ditto the bare literals vs. macros. I have filed a ticket for this:
https://jira.mongodb.org/browse/CDRIVER-4280